### PR TITLE
front-port v1.2.13 to develop

### PR DESCRIPTION
**Pull request by joe-bowman**
Front port of the code changes between v1.2.12 and v1.2.13.

### CODE REVIEW

File: x/interchainstaking/keeper/receipt.go [hunk 1]
```diff
@@ -25,49 +25,53 @@ const (
 	ICATimeout      = time.Hour * 6
 )
 
-func (k *Keeper) HandleReceiptForTransaction(ctx sdk.Context, txr *sdk.TxResponse, txn *tx.Tx, zone *types.Zone) error {
-	k.Logger(ctx).Info("deposit receipt.", "ischeck", ctx.IsCheckTx(), "isrecheck", ctx.IsReCheckTx())
-	hash := txr.TxHash
+func (k Keeper) HandleReceiptTransaction(ctx sdk.Context, txn *tx.Tx, txHash string, zone types.Zone) error {
```
Comment (aljo242) [inline x/interchainstaking/keeper/receipt.go:28]:
```suggestion
func (k *Keeper) HandleReceiptTransaction(ctx sdk.Context, txn *tx.Tx, txHash string, zone *types.Zone) error {
```
I think we should keep these

File: x/interchainstaking/keeper/delegation.go [hunk 2]
```diff
@@ -280,28 +280,34 @@ func (k *Keeper) MakePerformanceDelegation(ctx sdk.Context, zone *types.Zone, va
...
+			pendingAmount = pendingAmount.Add(receiptInfo.Amount...)
 		}
 		return false
 	})
 
-	coins := sdk.NewCoins(delAddrBalance).Sub(amount...)
+	var hasNeg bool
 
-	if coins.IsAnyNegative() || coins.IsZero() {
+	coinsToFlush, hasNeg = sdk.NewCoins(delAddrBalance).SafeSub(pendingAmount...)
```
Comment (aljo242) [inline x/interchainstaking/keeper/delegation.go:295]: We can just make this 

```go
	coinsToFlush, hasNeg := sdk.NewCoins(delAddrBalance).SafeSub(pendingAmount...)
```

and remove the `var` declarations

File: x/interchainstaking/keeper/receipt.go [hunk 3]
```diff
@@ -25,49 +25,53 @@ const (
...
+	k.Logger(ctx).Info("Deposit receipt.", "ischeck", ctx.IsCheckTx(), "isrecheck", ctx.IsReCheckTx())
+	hash := txHash
```
Comment (muku314115) [inline x/interchainstaking/keeper/receipt.go:30]: Can just use the txHash directly in the function or rename it to hash as it is only being read

File: proto/quicksilver/interchainquery/v1/query.proto [hunk 4]
```diff
@@ -29,16 +30,17 @@ message QueryRequestsRequest {
 // QueryParamsResponse is the response type for the Query/Params RPC method.
 message QueryRequestsResponse {
   // params defines the parameters of the module.
-  repeated quicksilver.interchainquery.v1.Query queries = 1 [(gogoproto.nullable) = false];
+  repeated quicksilver.interchainquery.v1.Query queries = 1
+      [ (gogoproto.nullable) = false ];
```
Comment (ajansari95) [inline proto/quicksilver/interchainquery/v1/query.proto:34]: why new line?

**Review by ajansari95 — COMMENTED** [review]
just a single nit in addition to @muku314115  and @aljo242

File: proto/quicksilver/interchainquery/v1/query.proto (hunk 4 above)
Comment (aljo242) [reply proto/quicksilver/interchainquery/v1/query.proto:34]: i think just formatted

File: x/interchainstaking/keeper/receipt.go (hunk 3 above)
Comment (joe-bowman) [reply x/interchainstaking/keeper/receipt.go:30]: Done in #445

File: x/interchainstaking/keeper/delegation.go (hunk 2 above)
Comment (joe-bowman) [reply x/interchainstaking/keeper/delegation.go:295]: resolved in #445

File: x/interchainstaking/keeper/receipt.go (hunk 1 above)
Comment (joe-bowman) [reply x/interchainstaking/keeper/receipt.go:28]: Resolved in #445

**Comment by joe-bowman** [comment]
Closed in favour of #445 as that is where all the changes have been made.

**Pull request closed without merging**